File: src/main/java/korzeniowski/mateusz/rpg/combat/manager/model/unit/UnitTemplate.java
package korzeniowski.mateusz.rpg.combat.manager.model.unit;

import korzeniowski.mateusz.rpg.combat.manager.model.skill.Skill;
import korzeniowski.mateusz.rpg.combat.manager.model.talent.Talent;
import korzeniowski.mateusz.rpg.combat.manager.model.talent.TalentTemplate;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document(collection = "units")
@Data
public class UnitTemplate {

    @Id
    private String id;
    @Indexed(unique = true)
    private String name;
    private List<Attribute> attributes;
    private List<Skill> skills;
    private List<Talent> talents;
}

File: rpg-combat-frontend/src/model/unitTemplate.ts
export interface Attribute {
    key: string;
    value: number;
}

export interface Skill {
    id: string;
    name: string;
    relatedAttribute: string;
    advances: number;
}

export interface Talent {
    id: string;
    name: string;
}

export interface UnitTemplateDto {
    id: string;
    name: string;
    attributes: Attribute[];
    skills: Skill[];
    talents: Talent[];
}

export const emptyUnitTemplate: UnitTemplateDto = {
    id: "",
    name: "",
    attributes: [],
    skills: [],
    talents: []
};